fe/fe-core/src/main/java/com/starrocks/lake/compaction/CompactionScheduler.java

```diff
@@ -452,6 +452,22 @@ public void cancelCompaction(long txnId) {
         }
     }
 
+    protected ConcurrentHashMap<PartitionIdentifier, CompactionJob> getRunningCompactions() {
+        return runningCompactions;
+    }
+
+    public boolean existCompaction(long txnId) {
+        for (Iterator<Map.Entry<PartitionIdentifier, CompactionJob>> iterator = getRunningCompactions().entrySet().iterator();
```

Comment:
Why does map need to be completely iterated?

Reply:
txnId is not map key